longest_common_subsequence/myers.py

```diff
@@ -0,0 +1,151 @@
+from math import ceil
+
+class EditBox:
+  def __init__(self, text_a, text_b, a_start, a_end, b_start, b_end):
+    self.text_a = text_a
+    self.text_b = text_b
+    self.a_start = a_start
+    self.a_end = a_end
+    self.b_start = b_start
+    self.b_end = b_end
+
+  def len(self):
+    return (self.a_end - self.a_start, self.b_end - self.b_start)
+
+  def compare(self, x, y):
+    return self.text_a[x + self.a_start] == self.text_b[y + self.b_start]
+
+  def cut_start(self, x, y):
+    return EditBox(
+        self.text_a,
+        self.text_b,
+        self.a_start,
+        self.a_start + x,
+        self.b_start,
+        self.b_start + y
+    )
+
+  def cut_end(self, x, y):
+    return EditBox(
+        self.text_a,
+        self.text_b,
+        self.a_start + x,
+        self.a_end,
+        self.b_start + y,
+        self.b_end
+    )
+
+  def get_first_text(self, start, end):
+    return self.text_a[self.a_start + start: self.a_start + end + 1]
+
+  def get_second_text(self):
+    return self.text_b[self.b_start + 1:self.b_end + 1]
+
+def _walk_forward(edit_box, d, v_forward, v_backward):
+  (a_len, b_len) = edit_box.len()
+  delta = a_len - b_len
+
+  for k in range(-d, d + 1, 2):
+    if k == -d or (k != d and v_forward[k - 1] < v_forward[k + 1]):
+      x = v_forward[k + 1]
+    else:
+      x = v_forward[k - 1] + 1
+
+    y = x - k
+
+    start = (x,y)
+    while x < a_len and y < b_len and edit_box.compare(x + 1, y + 1):
+      x, y = x + 1, y + 1
+    v_forward[k] = x
+
+    if (delta % 2 == 1 and
+        delta -(d - 1) <= k <= delta + (d - 1) and
+        y >= v_backward[k - delta]):
+      return (start, (x, y))
+
+  return None
+
+def _walk_backward(edit_box, d, v_forward, v_backward):
+  (a_len, b_len) = edit_box.len()
+  delta = a_len - b_len
+
+  for k in range(-d, d + 1, 2):
+    back_k = k + delta
+    if k == -d or (k != d and v_backward[k - 1] > v_backward[k + 1]):
+      y = v_backward[k + 1]
+    else:
+      y = v_backward[k - 1] - 1
+    x = y + back_k
+
+    start = (x,y)
+    while x > 0 and y > 0 and edit_box.compare(x, y):
+      x, y = x - 1, y - 1
+    v_backward[k] = y
+
+    if (delta % 2 == 0 and
+        -d <= back_k <= d and
+        x <= v_forward[back_k]):
+      return ((x, y), start)
+
+  return None
+
+def _find_middle_snake(edit_box):
+  (a_len, b_len) = edit_box.len()
+  max_edit = ceil( a_len + b_len / 2)
```

Comment:
Check: should it be "ceil(a_len + b_len / 2)" or "ceil((a_len + b_len) / 2)"?

Reply:
It should be ceil((a_len + b_len) / 2), thanks.